tools/code-generation/generator/src/main/java/com/amazonaws/util/awsclientgenerator/generators/cpp/CppClientGenerator.java

```diff
@@ -666,6 +666,68 @@ protected Set<String> getOperationsToRemove(){
         return new HashSet<String>();
     }
 
+    protected SdkFileEntry generateClientSmithyHeaderFile(final ServiceModel serviceModel) {
+        Template template = velocityEngine.getTemplate("com/amazonaws/util/awsclientgenerator/velocity/cpp/smithy/SmithyClientHeader.vm", StandardCharsets.UTF_8.name());
+
+        VelocityContext context = createContext(serviceModel);
+        context.put("CppViewHelper", CppViewHelper.class);
+        context.put("RequestlessOperations", requestlessOperations);
+        context.put("AuthSchemeResolver", "SigV4AuthSchemeResolver");
```

Comment:
I think we may need to code gen AuthSchemeResolvers per client. i.e. from [smithy](https://smithy.io/2.0/spec/authentication-traits.html#auth-trait)

>11.7. auth trait
Summary
Defines the priority ordered authentication schemes supported by a service or operation. When applied to a service, it defines the default authentication schemes of every operation in the service. When applied to an operation, it defines the list of all authentication schemes supported by the operation, overriding any auth trait specified on a service.
Trait selector
:is(service, operation)
>Service or operation shapes
>Value type
This trait contains a priority ordered list of unique string values that reference authentication scheme shape IDs defined on a service shape.

from my understand the [auth list](https://github.com/aws/aws-sdk-cpp/blob/main/tools/code-generation/api-descriptions/dynamodb-2012-08-10.normal.json#L15) _is_ the scheme provider 

from the SRA book:
```
The AuthSchemeResolver accepts data from the request (e.g. request members) and client (e.g. AWS region) and returns a list of authentication scheme options that the SDK should use when authenticating that request. Each authentication scheme option includes the selected scheme ID and properties for use when generating the identity and request signature. The SDK will choose the first authentication scheme in the result list that it supports. An authentication scheme is considered “supported” if the SDK recognizes the selected scheme ID and the SDK has an identity resolver for the selected scheme ID. If the identity resolver fails after the scheme ID is chosen, the request fails and does not move on to the next authentication scheme.
```

so in my head the "AuthSchemeResolver" is this list and at run time we select the first one we support.


for now though since we are doing only one client -- i think we can hardcode it.